pkg/cliplugins/bind/plugin/bind_compute.go

```diff
@@ -0,0 +1,295 @@
+	if len(o.APIExports) == 0 {
+		o.APIExports = []string{
+			"root:compute:kubernetes",
+			fmt.Sprintf("%s:kubernetes", o.ComputeWorkspace.String()),
+		}
+	}
+
+	// select all ns if namespace selector is not set
+	if len(o.NamespaceSelector) == 0 {
+		o.NamespaceSelector = labels.Everything().String()
+	}
+
+	// select all locations is location selectos is not set
+	if len(o.LocationSelectors) == 0 {
+		o.LocationSelectors = []string{labels.Everything().String()}
+	}
+
+	return nil
+}
+
+// Validate validates the BindOptions are complete and usable.
+func (o *BindWorkloadOptions) Validate() error {
+	return nil
+}
+
+// Run create a placement in the workspace linkind to the compute workspace
+func (o *BindWorkloadOptions) Run(ctx context.Context) error {
+	config, err := o.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+	userWorkspaceKcpClient, err := kcpclient.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	// build config to connect to compute workspace
+	computeWorkspaceConfig := rest.CopyConfig(config)
+	url, _, err := helpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return err
+	}
+
+	url.Path = path.Join(url.Path, o.ComputeWorkspace.Path())
+	computeWorkspaceConfig.Host = url.String()
+	computeWorkspaceKcpClient, err := kcpclient.NewForConfig(computeWorkspaceConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	err = o.hasSupportedSyncTargets(ctx, computeWorkspaceKcpClient)
+	if err != nil {
+		return err
+	}
+
+	err = o.applyPlacement(ctx, userWorkspaceKcpClient)
+	if err != nil {
+		return err
+	}
+
+	err = o.applyAPIBinding(ctx, userWorkspaceKcpClient)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func apiBindingName(clusterName logicalcluster.Name, apiExportName string) string {
+	hash := sha256.Sum224([]byte(clusterName.Path()))
+	base36hash := strings.ToLower(base36.EncodeBytes(hash[:]))
+	return fmt.Sprintf("%s-%s", apiExportName, base36hash[:8])
+}
+
+func (o *BindWorkloadOptions) applyAPIBinding(ctx context.Context, client kcpclient.Interface) error {
+	apiBindings, err := client.ApisV1alpha1().APIBindings().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	desiredAPIExports := sets.NewString(o.APIExports...)
+	existingAPIExports := sets.NewString()
+	for _, binding := range apiBindings.Items {
+		if binding.Spec.Reference.Workspace == nil {
+			continue
+		}
+		existingAPIExports.Insert(fmt.Sprintf("%s:%s", binding.Spec.Reference.Workspace.Path, binding.Spec.Reference.Workspace.ExportName))
+	}
+
+	diff := desiredAPIExports.Difference(existingAPIExports)
+	var errs []error
+	for export := range diff {
+		lclusterName, name := logicalcluster.New(export).Split()
+		apiBinding := &apisv1alpha1.APIBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: apiBindingName(lclusterName, name),
+			},
+			Spec: apisv1alpha1.APIBindingSpec{
```

Comment:
Do we need to accept permission claims?

Reply:
I don't think so. I think that if we would need to accept permission claims, it would be done explicitly using the claim-related commands.